### PR TITLE
Feat: Using `Makefile` to build xray

### PR DESCRIPTION
**Pull request by chise0713**
Linked: #2879

### CODE REVIEW

**Comment by chise0713** [comment]
I didn't modify any Go files or `test.yml` , so these failed tests may be due to some error.

File: README.md [hunk 1]
```diff
@@ -128,13 +128,7 @@
 ### Windows
```
Comment (yuhan6665) [inline README.md:128]: We can clear this line as well

**Comment by yuhan6665** [comment]
感谢佬 有一个comment
我们需要测试一下 ~~要发一个新版~~

**Comment by BI7PRK** [comment]
现在温饱就思淫欲了, 我要感谢xray ^_^

**Comment by yuhan6665** [comment]
感谢佬！

> 现在温饱就思淫欲了, 我要感谢xray ^_^

淫欲莫非是 ~~Dire Straits~~

**Comment by KobeArthurScofield** [comment]
将编译部分独立出来的确灵活了很多

~~但是因为文件名的原因自带容易被不明真相优化的光环~~

**Pull request merged**